core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -757,11 +758,12 @@ private[spark] class Master(
       case fnf: FileNotFoundException =>
         // Event logging is enabled for this application, but no event logs are found
         val title = s"Application history not found (${app.id})"
-        var msg = s"No event logs found for application $appName in $eventLogFile."
-        logWarning(msg)
+        var msg = s"No event logs found for application $appName in ${app.desc.eventLogDir}."
+        val exception = URLEncoder.encode(Utils.exceptionString(fnf), "UTF-8")
+        logWarning(msg, fnf)
```

Comment:
as commented elsewhere, I don't think logging the stack trace here is particularly useful